## Data Analysis Using SQL and Pandas on **Chinkook Database**

### Importing Libraries

In [87]:
import pandas as pd
import sqlite3

### Connecting To my Sqlite file


In [88]:
conn = sqlite3.connect('Chinook_Sqlite.sqlite')
cursor = conn.cursor()

### Sanity Checking That All Tables are Loaded

In [89]:
pd.read_sql_query("SELECT name FROM sqlite_master WHERE type = 'table';",conn)

,name
0,Album
1,Artist
2,Customer
3,Employee
4,Genre
5,Invoice
6,InvoiceLine
7,MediaType
8,Playlist
9,PlaylistTrack


### Top **5** Genres With the Most Revenue

In [90]:
pd.read_sql_query("SELECT g.GenreId,g.Name,SUM(il.UnitPrice*il.Quantity) AS total_revenue FROM Genre g JOIN Track t ON g.GenreId = t.GenreId JOIN InvoiceLine il ON il.TrackId = t.TrackId GROUP BY g.GenreID,g.Name ORDER BY total_revenue DESC LIMIT 5",conn)

,GenreId,Name,total_revenue
0,1,Rock,826.65
1,7,Latin,382.14
2,3,Metal,261.36
3,4,Alternative & Punk,241.56
4,19,TV Shows,93.53


##### This Table shows us that **Rock** has the highest revenue of **826.65 dollars**.

### Top **5** customers by Total Amount Spent

In [91]:
pd.read_sql_query("SELECT c.CustomerId,c.FirstName,c.LastName,SUM(i.Total) AS Total_Amount_Spent FROM Customer c JOIN Invoice i ON c.CustomerId = i.CustomerId GROUP BY c.CustomerId,c.FirstName,c.LastName ORDER BY Total_Amount_Spent DESC LIMIT 5",conn)

,CustomerId,FirstName,LastName,Total_Amount_Spent
0,6,Helena,Holý,49.62
1,26,Richard,Cunningham,47.62
2,57,Luis,Rojas,46.62
3,45,Ladislav,Kovács,45.62
4,46,Hugh,O'Reilly,45.62


##### CustomerId's **6, 26, 57, 45, 46** spent the most Total Amount with *Helena Holý* spent the most amount of **49.62**

### Artist with the most tracks

In [92]:
pd.read_sql_query("SELECT Artist.ArtistId,Artist.Name,COUNT(Track.TrackId) AS Total_Tracks FROM Artist JOIN Album ON Artist.ArtistId = Album.ArtistId JOIN Track ON Track.AlbumId = Album.AlbumId GROUP BY Artist.ArtistId,Artist.Name ORDER BY Total_Tracks DESC LIMIT 1",conn)

,ArtistId,Name,Total_Tracks
0,90,Iron Maiden,213


#### **Iron Maiden** is the artist with the most number of tracks of *213*